code/modules/events/earthquake.dm

```diff
@@ -0,0 +1,174 @@
+	var/turf/low_midpoint = TURF_MIDPOINT(fracture_point_low, epicenter)
+
+	// We populate a list with the midpoints and midpoints of midpoints to create a rough line of turfs to compare distances against.
+	var/list/turfs_to_compare = list(
+			fracture_point_high,
+			fracture_point_low,
+			high_midpoint,
+			low_midpoint,
+			TURF_MIDPOINT(fracture_point_high, high_midpoint),
+			TURF_MIDPOINT(fracture_point_low, low_midpoint),
+			TURF_MIDPOINT(high_midpoint, epicenter),
+			TURF_MIDPOINT(low_midpoint, epicenter),
+		)
+
+	// Find the shortest distance between each turf in the list and the rough fault line we've just established
+	for(var/turf/turf_to_check in turfs_to_shred)
+		var/nearest_distance = get_dist(turf_to_check, epicenter)
+
+		for(var/turf/turf_to_compare in turfs_to_compare)
+			nearest_distance = min(get_dist(turf_to_check, turf_to_compare), nearest_distance)
+
+		// If the turf is too far from any point on our fault line estimate, we remove it. If it's on the edge, we lightly damage it
+		if(nearest_distance > 2)
+			if(nearest_distance == 3)
+				edges += turf_to_check
+			turfs_to_shred -= turf_to_check
+
+	// Grab a list of turfs below the ones we're going to destroy.
+	// If we're at the bottom layer, it will just tear up the flooring a bunch (exposing it to LAVA).
+	for(var/turf/turf_to_quake in turfs_to_shred)
+		underbelly += SSmapping.get_turf_below(turf_to_quake)
+
+/datum/round_event/earthquake/announce(fake)
+	priority_announce("Planetary monitoring systems indicate a devastating seismic event in the near future.", "Seismic Report")
+
+/datum/round_event/earthquake/start()
+	notify_ghosts("The earthquake's epicenter has been located: [get_area_name(epicenter)]!", source = epicenter, header = "Rumble Rumble Rumble!")
+
+/datum/round_event/earthquake/tick()
+	if(ISMULTIPLE(activeFor, 5))
+		for(var/turf/turf_to_quake in turfs_to_shred)
+			turf_to_quake.Shake(0.1, 0.1, 1 SECONDS)
+
+		if(ISMULTIPLE(activeFor, 10))
+			for(var/mob/earthquake_witness as anything in GLOB.player_list)
+				if(!is_station_level(earthquake_witness.z))
+					continue
+				shake_camera(earthquake_witness, 1 SECONDS, 1 + (activeFor % 10))
+				earthquake_witness.playsound_local(
+					earthquake_witness,
+					pick(
+						'sound/misc/earth_rumble_distant1.ogg',
+						'sound/misc/earth_rumble_distant2.ogg',
+						'sound/misc/earth_rumble_distant3.ogg',
+						'sound/misc/earth_rumble_distant4.ogg',
+					),
+					75,
+				)
+
+			for(var/turf/turf_to_quake in underbelly)
+				turf_to_quake.Shake(0.1, 0.1, 1 SECONDS)
+
+	if(activeFor == end_when - 2)
+		for(var/turf/turf_to_quake in turfs_to_shred)
+			turf_to_quake.Shake(0.5, 0.5, 1 SECONDS)
+			for(var/mob/living/carbon/quake_victim in turf_to_quake)
+				quake_victim.Knockdown(7 SECONDS)
+				quake_victim.Paralyze(5 SECONDS)
+				if(quake_victim.client)
+					quake_victim.client.give_award(/datum/award/achievement/misc/earthquake_victim, quake_victim)
```

Comment:
I was already considering moving it to when the chasm actually forms (and the person earning the achievement is absolutely 100% going to die), since there's a few situations I can envision where someone could survive/not fall after earning it. I'll probably just do that.